ring-core/test/ring/core/test/protocols.clj

```diff
@@ -53,3 +54,36 @@
           response {:body nil}]
       (write-body-to-stream (:body response) response output)
       (is (= "" (.toString output))))))
+
+(defn output-stream-with-close-flag []
+  (let [stream-closed? (atom false)
+        output-stream  (proxy [OutputStream] []
+                         (write
+                           ([])
+                           ([^bytes _])
+                           ([^bytes _ _ _]))
+                         (close [] (reset! stream-closed? true)))]
+    [output-stream stream-closed?]))
+
+(defn error-input-stream []
+  (proxy [InputStream] []
+    (read
+      ([] (throw (IOException. "test error")))
+      ([^bytes _] (throw (IOException. "test error")))
+      ([^bytes _ _ _] (throw (IOException. "test error"))))))
+
+(deftest test-error-when-writing-body
+  (testing "input streams with error"
+    (let [[output stream-closed?] (output-stream-with-close-flag)
+          response                {:body (SequenceInputStream.
+                                           (io/input-stream (io/resource "ring/assets/hello world.txt"))
```

Comment:
Please ensure lines do not exceed 80 characters. See the contributing guidelines for more information.

Reply:
Changed code a little to fit in 80 characters.
Updated test/ring/adapter/test/jetty.clj to fit maximum line length.